### PR TITLE
Minor: add the concise way for matching numerics

### PR DESCRIPTION
**Pull request by izveigor**
# Rationale for this change
The array "NUMERICS", that are defined in the file "datafusion/expr/src/type_coercion/aggregates.rs" (see https://github.com/apache/arrow-datafusion/blob/main/datafusion/expr/src/type_coercion/aggregates.rs#L30-L41) contains all types, which found in many "match" and "matches!" cases.

I think this PR suggests more concise way for these cases.

# What changes are included in this PR?
Rewritten "match" statements and "matches!" macros.

# Are these changes tested?
Yes

# Are there any user-facing changes?
No

### CODE REVIEW

File: datafusion/expr/src/type_coercion/aggregates.rs [hunk 1]
```diff
@@ -314,77 +314,45 @@ pub fn sum_return_type(arg_type: &DataType) -> Result<DataType> {
...
-        | DataType::UInt8
-        | DataType::UInt16
-        | DataType::UInt32
-        | DataType::UInt64
-        | DataType::Float32
-        | DataType::Float64 => Ok(DataType::Float64),
-        other => Err(DataFusionError::Plan(format!(
-            "VAR does not support {other:?}"
-        ))),
+    if NUMERICS.contains(&arg_type) {
```
Comment (alamb) [inline datafusion/expr/src/type_coercion/aggregates.rs:317]: 🤔  this is very close to https://docs.rs/arrow/34.0.0/arrow/datatypes/enum.DataType.html#method.is_numeric except that is_numeric also includes Decimals.

**Review by alamb — COMMENTED** [review]
Looks like a nice cleanup to me -- thank you @izveigor

**Review by alamb — APPROVED** [review]
I also verified that NUMERICS contains the same values 
https://github.com/apache/arrow-datafusion/blob/8c34ca4fa34787b137b48ce4f6ffd41b64a1a633/datafusion/expr/src/type_coercion/aggregates.rs#L30-L41

File: datafusion/expr/src/type_coercion/aggregates.rs [hunk 2]
```diff
@@ -314,77 +314,45 @@ pub fn sum_return_type(arg_type: &DataType) -> Result<DataType> {
...
-        | DataType::UInt8
-        | DataType::UInt16
-        | DataType::UInt32
-        | DataType::UInt64
-        | DataType::Float32
-        | DataType::Float64 => Ok(DataType::Float64),
-        other => Err(DataFusionError::Plan(format!(
-            "VAR does not support {other:?}"
-        ))),
+    if NUMERICS.contains(arg_type) {
```
[the remark this answers is not in the document]
Comment (alamb) [reply datafusion/expr/src/type_coercion/aggregates.rs:317]: I think the `DataType::is_numeric` also includes `Decimals` 🤔

[the remark this answers is not in the document]
Comment (comphead) [reply datafusion/expr/src/type_coercion/aggregates.rs:317]: Right, perhaps we can do a helper method checking is_numeric and exclude decimals.

[the remark this answers is not in the document]
Comment (comphead) [reply datafusion/expr/src/type_coercion/aggregates.rs:317]: sort of
```
fn is_numeric(arg_type: &DataType) -> bool {
    return arg_type.is_numeric() && match arg_type {
        Decimal128(_, _) | Decimal() => false
        _ => true
    } 
}
```

**Review by jackwener — APPROVED** [review]
Make sense to me. A good job to cleanup

**Review by yjshen — APPROVED** [review]

**Pull request merged**